### PR TITLE
Feaure/metrics docs

### PR DESCRIPTION
**Pull request by jcampbell**
What's here:
- Changes to get_metric API (internal) and requested_metrics configuration (with clear demarcation as beta)
- Docs for Metrics
- Glossary
- Rename StoreAction to StoreValidationResultAction
- Improve doctest examples

### CODE REVIEW

File: docs/features/metrics.rst [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+##############
+
+Metrics are values derived from one or more Data Assets that can be used to evaluate expectations or to summarize the
+result of Validation. A Metric is obtained from an ExpectationValidationResult or ExpectationSuiteValidationResult by
+providing the `metric_name` and `metric_kwargs` or `metric_kwargs_id`.
+
+A metric name is a dot-delimited string that identifies the value, such as `expect_column_values_to_be_unique
+.success` or `expect_column_values_to_be_between.result.unexpected_percent`.
+
+Metric Kwargs are key-value pairs that identify the within the context of the validation, such as "column": "Age".
```
Comment (eugmandel) [inline docs/features/metrics.rst]: "identify the metric"?

File: docs/features/metrics.rst [hunk 2]
```diff
@@ -0,0 +1,24 @@
+.. _metrics:
+
+##############
+Metrics
+##############
+
+Metrics are values derived from one or more Data Assets that can be used to evaluate expectations or to summarize the
```
Comment (eugmandel) [inline docs/features/metrics.rst:7]: Are we still using "Data Asset" and "Batch" interchangeably?

File: docs/tutorials/saving_metrics.rst [hunk 3]
```diff
@@ -0,0 +1,79 @@
+.. _saving_metrics:
+
+###############
+Saving Metrics
+###############
+
+Saving metrics during Validation makes it easy to construct a dashboard create a new data series based on observed
```
Comment (eugmandel) [inline docs/tutorials/saving_metrics.rst]: "a dashboard create" -> "a dashboard that creates" ?

**Review by eugmandel — CHANGES_REQUESTED** [review]

File: docs/features/metrics.rst (hunk 1 above)
Comment (jcampbell) [reply docs/features/metrics.rst]: fixed!

File: docs/features/metrics.rst (hunk 2 above)
Comment (jcampbell) [reply docs/features/metrics.rst:7]: I'm trying to migrate to batch == data, data asset == "validatable"...should this be changed?

File: docs/tutorials/saving_metrics.rst (hunk 3 above)
Comment (jcampbell) [reply docs/tutorials/saving_metrics.rst]: Fixed!

File: docs/tutorials/saving_metrics.rst [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+
+The next time your DataContext is loaded, it will connect to the database and initialize a table to store metrics if
+one has not already been created. See the :ref:`metrics_reference` for more information on additional configuration
+options.
+
+*********************************
+Configuring a Validation Action
+*********************************
+
+Once a MetricStore is available, it is possible to configure a a new `StoreMetricsAction` to save metrics during
```
Comment (eugmandel) [inline docs/tutorials/saving_metrics.rst]:
```suggestion
Once a MetricStore is available, it is possible to configure a new `StoreMetricsAction` to save metrics during
```

**Review by eugmandel — CHANGES_REQUESTED** [review]

**Review by eugmandel — APPROVED** [review]

**Pull request merged**